SystemsGenetics/GEMmaker
# Adds support for multiqc summary reports

## Pull Request Details
This branch adds support for multiqc (http://multiqc.info/).  Thus, when a GEMmaker run is completely finished the user can use multiqc to generate a very useful report summarizing the output.  The following changes were made.  This branch also improves support for trimmomatic by adding a FASTA file of adapters.  This way, we no longer need to require that the user set the illumina clip path.

1) The Hisat2 script is changed to generate a stats report. This report is now published in the sample directory.
2)  The trimmomatic STDOUT/STDERR is redirected to a log file which is now published in the sample directory.
3)  A new samtools stats command was added to the samtools_index process.  The STDOUT is saved to a file which is then published in the same directory.
4)  A new multiqc_config.yaml file has been added to the root directory.  This ensures that GEMmaker branding and introduction are part of the final report.
5)  A new GEMmaker logo has been added and all images (those that previously existed) are now in a new images folder.  ![GEMmaker Logo](https://raw.githubusercontent.com/SystemsGenetics/GEMmaker/reporting/images/GEMmaker-logo-sm.png)
6) The README.md has been updated accordingly.
7)  A new files directory has been added and the fasta_adapter.txt file has been added.  The path to this file has now been set in the nextflow.config.example file.  User's no longer need to change it unless they want to use a different trimming file.  This also means that you no longer need to load trimmomatic on Kamiak prior to running nextflow as it no longer requires the environment variable.
8) Made a fix to the fpkm2gem.sh script for finding all FPKM directories (even in the nested directory structure)

## How to test
Try this on a brand new project, or you have to remember to update the clip path for trimmomatic in your existing project.
1) Install MultiQC (http://multiqc.info/docs/#installing-multiqc).  I used the following on Kamiak: ```module add python3; pip install --user multiqc```.
2) Pick a project to test this on.  I haven't not yet tested the example data to see what the report looks like but I think it should work.  I suspect that trimmomatic will not find anything to trim with the example data, but I may be wrong.
3)  Switch to the reporting branch.
4)  Fix the trimmomatic clip path in your nextflow.config  (see the nextflow.config.example for what this should look like)
5)  Run GEMmaker
6)  After the workflow completes, run multiqc: ```multiqc . ```
7) You should see a new multiqc_report.html file and a multiqc_data directory.  Download both to your computer and you can open the report in a web browser.  Note: if you have too many samples then some of the interactive features of the reports won't work, but this is normal for multiqc.
